src/server/handlers/store.js

```diff
@@ -0,0 +1,42 @@
+import 'isomorphic-fetch';
+import url from 'url';
+
+import { conf } from '../helpers/config';
+
+const STORE_CPI_API_URL = conf.get('STORE_CPI_API_URL');
+
+export const getSnapDetails = async (req, res) => {
+  const snapName = req.params.name;
+  const query = {};
+
+  // pass allowed query parameters
+  for (const key of ['fields', 'channel', 'confinement'] ) {
+    if (req.query && req.query.hasOwnProperty(key)) {
+      query[key] = req.query[key];
+    }
+  }
```

Comment:
A more concise approach might be to construct the query string like this:

    const query = qs.stringify(req.query, {
      filter: ['fields', 'channel', 'confinement']
    });

(See `src/server/handlers/login.js`, where we do this.)

Also, do you actually need `confinement`?  It doesn't seem like something that BSI would need, and the current search.apps.ubuntu.com codebase doesn't support it yet (there's a TODO comment for it).